src/sortStudents.ts

```diff
@@ -1,16 +1,65 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsList: Student[] = [...students];
+  const getAverageGrade = (student: Student): number => {
+    const gradesSum = student.grades.reduce((sum, grade) => sum + grade, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    return gradesSum / student.grades.length;
+  };
+  let sortFunction: (a: Student, b: Student) => number;
```

Comment:
Better use meaningful variable names.
```suggestion
  let sortFunction: (firstStudent: Student, secondStudent: Student) => number;
```